buildSrc/src/main/java/io/vantiq/extsrc/assy/tasks/AssemblyGen.java

```diff
@@ -0,0 +1,336 @@
+    
+    public static final String KAMELETS_RESOURCE_FOLDER = "kamelets";
+    public static final String KAMELETS_RESOURCE_PATH = KAMELETS_RESOURCE_FOLDER + "/";
+    public static final String YAML_FILE_TYPE = ".yaml";
+    public static final String YML_FILE_TYPE = ".yml";
+    public static final String SOURCE_KAMELET_PREFIX = "-source.kamelet";
+    public static final String SINK_KAMELET_PREFIX = "-sink.kamelet";
+    public static final String SOURCE_KAMELET_YAML_SUFFIX = SOURCE_KAMELET_PREFIX + YAML_FILE_TYPE;
+    public static final String SOURCE_KAMELET_YML_SUFFIX = SOURCE_KAMELET_PREFIX + YML_FILE_TYPE;
+    public static final String SINK_KAMELET_YAML_SUFFIX = SINK_KAMELET_PREFIX + YAML_FILE_TYPE;
+    public static final String SINK_KAMELET_YML_SUFFIX = SINK_KAMELET_PREFIX + YML_FILE_TYPE;
+    
+    public final Project project;
+    
+    @Inject
+    public AssemblyGen(Project project) {
+        if (project == null) {
+            throw new GradleException("AssemblyGen requires a non-null project.");
+        }
+        this.project = project;
+    }
+    
+    @OutputDirectory
+    public File getOutputDirectory() {
+        if (!(project.property("buildDir") instanceof File)) {
+            throw new GradleException("Internal Error: buildDir not a file: " + project.property("buildDir"));
+        }
+    
+        File buildDir = (File) project.property("buildDir");
+    
+        assert buildDir != null;
+        return Path.of(buildDir.getAbsolutePath(), KAMELET_GEN_BASE).toFile();
+    }
+    
+    /**
+     * Overall processing for the generation of assemblies from kamelet declarations.
+     *
+     * @throws Exception
+     */
+    @TaskAction
+    public void generateAssemblies() throws Exception {
+        File buildDir = (File) project.property("buildDir");
+        assert buildDir != null;
+        if (!(buildDir.exists() && buildDir.isDirectory())) {
+            throw new GradleException("Internal Error: BuildDir not present or not directory: " + buildDir);
+        }
+        Path generateBase = Paths.get(buildDir.getAbsolutePath(), KAMELET_GEN_BASE);
+        if (!Files.exists(generateBase)) {
+            log.info("Creating base dir: {}", generateBase);
+            generateBase = Files.createDirectories(generateBase);
+        }
+        LoadSettings settings = LoadSettings.builder().build();
+        Load load = new Load(settings);
+    
+        // From the gradle data, construct the list of resolved dependencies & build a class loader to use
+        Configuration conf = project.getConfigurations().getByName("runtimeClasspath");
+        Set<File> files = conf.resolve();
+        log.info("File count from runtimeClasspath: {}", files.size());
+        File[] jars = files.toArray(new File[0]);
+        ArrayList<URL> urls = new ArrayList<>(jars.length);
+        for (File f: jars) {
+            log.debug("Adding {} to list of URLs", f.toURI().toURL());
+            urls.add(f.toURI().toURL());
+        }
```

Comment:
The class loader below is constructed to load the dependencies as declared in the gradle file.  We use this to obtain the kamelet definitions.